### PR TITLE
[resol] Fix handling of WeekTime fields on vbus

### PR DESCRIPTION
**Pull request by ramack**
Controllers (for example the SKSR 1/2/3) on the VBUS transmitting fields of type WeekTime did not work as decribed in #13447 which is fixed by this PR.

The resol-vbus-java library used by this binding was also improved for this controller in the meantime, that's why this PR integrates an upgrade to the latest version of this lib.

A compiled version of this code change can be found [here](https://github.com/ramack/openhab-addons/blob/51dd67d764f11084cb904aeabf7a02cb52f4791e/bundles/org.openhab.binding.resol/target/org.openhab.binding.resol-3.4.0-SNAPSHOT.jar?raw=true) and @paicl01 tested it successfully.

### CODE REVIEW

**Comment by lolodomo** [comment]
This looks like a breaking change to be mentioned in the release notes.

**Comment by ramack** [comment]
In a sense I agree, but as it breaks only for devices using the WeekTime type before it doesn't matter, as those did not work at all :smile:

**Comment by lolodomo** [comment]
But users will have to change their item type to make it work. So it is better to inform them.

**Comment by ramack** [comment]
yea why not. Where do I need to put it to make it occur in the release notes? The Readme in the binding doesn't seem to be the best place - on https://www.openhab.org/docs/developer/bindings/ I did not find anything and I am simply not contributing often enough to know it...

File: bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java [hunk 1]
```diff
@@ -249,11 +249,12 @@ protected void packetReceived(Specification spec, Language lang, Packet packet)
...
                                 } else {
                                     try {
-                                        QuantityType<?> q = new QuantityType<>(str);
+                                        QuantityType<?> q = new QuantityType<>(str, Locale
+                                                .getDefault()); /* vbus library returns the value in default locale */
                                         this.updateState(channelId, q);
                                     } catch (IllegalArgumentException e) {
                                         logger.debug("unit of '{}' unknown in openHAB", str);
-                                        QuantityType<?> q = new QuantityType<>(dd.toString());
+                                        QuantityType<?> q = new QuantityType<>(dd.toString(), Locale.getDefault());
```
Comment (lolodomo) [inline bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:257]: `dd` is a `Double`. I don't think you need to pass the locale there.

You are creating a `QuantityType` without any unit ?? That looks strange to me.

File: bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java [hunk 2]
```diff
@@ -287,13 +287,15 @@ protected void packetReceived(Specification spec, Language lang, Packet packet)
...
                             ResolBridgeHandler handler = (ResolBridgeHandler) b.getHandler();
                             String value;
+                            Locale loc;
                             if (handler != null) {
-                                value = pfv.formatTextValue(pfv.getPacketFieldSpec().getUnit(), handler.getLocale());
+                                loc = handler.getLocale();
                             } else {
-                                value = pfv.formatTextValue(pfv.getPacketFieldSpec().getUnit(), Locale.getDefault());
+                                loc = Locale.getDefault();
                             }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:295]: Your comment above: "vbus library returns the value in default locale".
So why not just using there like above `Locale.getDefault()` ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
> yea why not. Where do I need to put it to make it occur in the release notes? The Readme in the binding doesn't seem to be the best place - on https://www.openhab.org/docs/developer/bindings/ I did not find anything and I am simply not contributing often enough to know it...

No problem, I will guide you once the PR is merged. This will be added in a second step, outside the current PR.

File: bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:257]: Ok, you are passing a string without any unit and a locale. I don't know what will be the unit considered by the core framework in that case.

Comment (lolodomo) [reply bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:257]: I would have imagined to rather use in that case the constructor with a Number and a unit.

https://github.com/openhab/openhab-core/blob/4f35d474fc63f487b10042d05f01ea524fe312e8/bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityType.java#L164

Comment (ramack) [reply bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:257]: I would prefer that yes, but this code is only executed if the resol-vbus library returns a unit that is not known by openHAB. So I cannot convert it to a reasonable unit. I actually don't remember whether it commonly happens or whether it is only for a case that could theoretically. In this PR I only added the locale to cover the localized decimal point correctly - which was a change in the lib between 0.7 and 0.10.

Comment (ramack) [reply bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:257]: ah no, dd is the Double, so after the toString the QuantityType should not need special handling of the decimal point. You're totally right, adding the locale here is useless and even wrong if the default local uses , as decimal separator.

Would 
`QuantityType<?> q = new QuantityType<>(dd, AbstractUnit.ONE);`
be better?

File: bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:296]: By the way, you changed nothing there, so let's close my comment.

File: bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java [hunk 3]
```diff
@@ -249,11 +250,12 @@ protected void packetReceived(Specification spec, Language lang, Packet packet)
...
                                 } else {
                                     try {
-                                        QuantityType<?> q = new QuantityType<>(str);
+                                        QuantityType<?> q = new QuantityType<>(str, Locale
+                                                .getDefault()); /* vbus library returns the value in default locale */
                                         this.updateState(channelId, q);
                                     } catch (IllegalArgumentException e) {
                                         logger.debug("unit of '{}' unknown in openHAB", str);
-                                        QuantityType<?> q = new QuantityType<>(dd.toString());
+                                        QuantityType<?> q = new QuantityType<>(dd, AbstractUnit.ONE);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java:258]: Please rather use `Units.ONE`, importing Units from org.openhab.core.library.unit.Units

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM, thank you.

**Comment by lolodomo** [comment]
@ramack : please create a PR to explain the breaking change.
Here is an example of what to do (only one line to add in the same file):
https://github.com/openhab/openhab-distro/pull/1420

**Comment by ramack** [comment]
> @ramack : please create a PR to explain the breaking change. Here is an example of what to do (only one line to add in the same file): [openhab/openhab-distro#1420](https://github.com/openhab/openhab-distro/pull/1420)
like this?
https://github.com/openhab/openhab-distro/pull/1427

**Pull request merged**